src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/WaySectionProcessor.java

```diff
@@ -0,0 +1,906 @@
+                line.getIdentifier());
+
+        // Determines if we need to reverse the polyline and if a reverse edge is needed
+        final PbfOneWay oneWay = PbfOneWay.forTag(line);
+        final boolean hasReverseEdge = oneWay == PbfOneWay.NO;
+        final boolean isReversed = oneWay == PbfOneWay.REVERSED;
+
+        // Keep track of the start index, start/end nodes
+        int startIndex;
+        Optional<TemporaryNode> startNode = Optional.empty();
+        Optional<TemporaryNode> endNode = Optional.empty();
+
+        try
+        {
+            // Find the first node
+            startIndex = 0;
+            startNode = nodesToSectionAt.getNode(polyline.first());
+            if (!startNode.isPresent())
+            {
+                // Weren't able to find the starting node, abort slicing
+                logger.error(
+                        "Could not find starting Node for Line {} during way-sectioning. Aborting!",
+                        line.getIdentifier());
+                return newEdgesForLine;
+            }
+
+            // We've already processed the starting node, so we start with the first index
+            for (int index = 1; index < polyline.size(); index++)
+            {
+                // Check to see if this location is a node
+                endNode = nodesToSectionAt.getNode(polyline.get(index));
+                if (endNode.isPresent())
+                {
+                    // We found the end node, create the edge. Note: using occurrence minus one
+                    // since PolyLine uses zero-based numbering. We are incrementing only the
+                    // start node occurrence, since the end node will either be used as a future
+                    // start node or be the end of the way, in which case we don't care.
+                    final int startOccurrence = nodesToSectionAt.getOccurrence(startNode.get()) - 1;
+                    nodesToSectionAt.incrementOccurrence(startNode.get());
+                    final int endOccurrence = nodesToSectionAt.getOccurrence(endNode.get()) - 1;
+
+                    // Build the underlying polyline and reverse it, if necessary
+                    final PolyLine rawPolyline = polyline.between(polyline.get(startIndex),
+                            startOccurrence, polyline.get(index), endOccurrence);
+                    final PolyLine edgePolyline = isReversed ? rawPolyline.reversed() : rawPolyline;
+
+                    newEdgesForLine.add(new TemporaryEdge(identifierFactory.nextIdentifier(),
+                            edgePolyline, line.getTags(), hasReverseEdge));
+
+                    // Increment starting pointers
+                    startIndex = index;
+                    startNode = endNode;
+                }
+            }
+        }
+        catch (final Exception e)
+        {
+            logger.error("Failed to way-section line {}", line.getIdentifier(), e);
```

Comment:
This could return a half populated map of newEdgesForLine.  Is it better to return nothing in this case? Same could happen to the ring case below.

Reply:
Good point!